### ISSUE TITLE
read error when path contains a space

### ISSUE THREAD

**Issue by adokter** [adokter/bioRad#162]
Docker dependent functions (`read_pvolfile` for NEXRAD, `nexrad_to_odim`, `calculate_profile`) give an error when the file path contains a space

```
> read_pvolfile("/my/path/with a space/polar_volume_file")
Converting using Docker...
docker: invalid reference format.
See 'docker run --help'.
 Show Traceback
 
 Rerun with Debug
 Error in nexrad_to_odim_tempfile(file, verbose = verbose, mount = mount) : 
  Failed to start vol2bird Docker container. In addition: Warning message:
In mount_docker_container(normalizePath(mount, winslash = "/")) :
  failed to mount /my/path/with a space/polar_volume_file ... Go to 'Docker -> preferences -> File Sharing' and add this directory (or its root directory) as a bind mounted directory
```

**Comment by adokter** [comment]
The [`docker run` command](https://github.com/adokter/bioRad/blob/0d3e019ebce1c38a8c910039ad80868b12b3e74f/R/docker.R#L97-L99) in `mount_docker_container` expects special characters and spaces to have a leading escape, i.e. like this:
```
 mount_docker_container("~/Dropbox/Eleonora\\'s\\ falcon")
```
however the `normalize_Path` function does not, therefore I get a warning when passing the correct patch string
```
> normalizePath("~/Dropbox/Eleonora\\'s\\ falcon")
[1] "/Users/amd427/Dropbox/Eleonora\\'s\\ falcon"
Warning message:
In normalizePath("~/Dropbox/Eleonora\\'s\\ falcon") :
  path[1]="/Users/amd427/Dropbox/Eleonora\'s\ falcon": No such file or directory
```
Solution is therefore on Mac/Linux is to
1) apply normalizePath first
2) to the output of 1) add leading `\\` to all special characters and spaces (find out how)

Likely different solution needed on Windows

**Comment by adokter** [comment]
@stijnvanhoey maybe you know this from the top of your head:

I need a function that escapes the white space and special characters in a path, exactly as you would  need in a shell

On a Mac the string `~/Dropbox/Eleonora's falcon/dir with whitespace/` should be translated into
`/Users/adriaan/Eleonora\\'s\\ falcon/dir\\ with\\ whitespace/` (need double backslashes in the string, in order to have a real backslash when I execute it with a `system()` command).

On Windows (I think)  `c:/Dropbox/Eleonora's falcon/dir with whitespace/` should be translated into
`"c:\Dropbox\Eleonora 's falcon\dir with whitespace"` (so with explicit quotes " in the string).

I've checked functionality of `normalizePath` and `shQuote` but can't get the escaping quite right.

**Comment by adokter** [comment]
This seems promising:
```
system(command="ls",input="/Users/amd427/Dropbox/Eleonora's falcon")
```

**Comment by stijnvanhoey** [comment]
Just a guess: Would using 'file.path' to create the path name not properly parse this?

**Comment by adokter** [comment]
No unfortunately not - to interact with the Docker daemon I have to parse a path via a `system()` call in R, as in 
```
result <- system(
      paste("docker run -v ",
        "CORRECTLY_ESCAPED_PATH_GOES_HERE",
        ":/data -d --name vol2bird adokter/vol2bird sleep infinity",
        sep = ""),ignore.stdout = TRUE)
```
`file.path` doesn't add the escapes that you need in a shell that is opened by the `system()` command ...

**Comment by adokter** [comment]
Closed by PR #174

**Comment by adokter** [comment]
This issue persists when using a local installation

**Comment by adokter** [comment]
`nexrad_to_odim()` fixed in [d123b94](https://github.com/adokter/bioRad/pull/506/commits/d123b947e281d1482d340217886ae22668aaa41e)
